src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Transition.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.models
+
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.CronSchedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.ScheduleParser
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class Transition(
```

Comment:
Let's create a Conditions data class to hold the conditions:
```
Transition
    state_name: String
    conditions: Conditions
```

```
Conditions
    index_age: String?
    doc_count: Long?
    size: String?
    cron: CronSchedule?
```

And then in the future if we support different operators and grouping of conditions we can further extend Conditions with `must`, `should`, `must_not` properties to follow Elasticsearch's convention

You'll need to update the mappings for this, perhaps disabling the conditions mapping since it will eventually be nested/dynamic based on users choice.